thentos-core/src/Servant/Missing.hs

```diff
@@ -41,6 +42,12 @@ data FormReqBody
 fromEnvIdentity :: Applicative m => Env Identity -> Env m
 fromEnvIdentity env = pure . runIdentity . env
 
+-- | FIXME: not sure this is legal.  do we need to make the IORef thread safe?  by wrapping it into
+-- an STRef?  but it seems like we are using it exactly like it is intended to.
```

Comment:
Also, the InternalState should be closed somewhere, using closeInternalState, or else we need to call runResourceT from time to time. Otherwise, no releasing of resources ever happens, because tempFileBackEnd doesn't do that for us. Oh, you actually mention that in the FIXME below.